test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -218,13 +281,35 @@ var (
 )
 
 // vSphereFinalizers maps VSphere infrastructure resource types to their expected finalizers.
-var vSphereFinalizers = map[string]func(types.NamespacedName) []string{
-	"VSphereVM":              func(_ types.NamespacedName) []string { return []string{infrav1.VMFinalizer} },
-	"VSphereClusterIdentity": func(_ types.NamespacedName) []string { return []string{infrav1.VSphereClusterIdentityFinalizer} },
-	"VSphereDeploymentZone":  func(_ types.NamespacedName) []string { return []string{infrav1.DeploymentZoneFinalizer} },
-	"VSphereMachine":         func(_ types.NamespacedName) []string { return []string{infrav1.MachineFinalizer} },
-	"IPAddressClaim":         func(_ types.NamespacedName) []string { return []string{infrav1.IPAddressClaimFinalizer} },
-	"VSphereCluster":         func(_ types.NamespacedName) []string { return []string{infrav1.ClusterFinalizer} },
+var vSphereFinalizers = func() map[string]func(types.NamespacedName) []string {
+	if testMode == SupervisorTestMode {
+		return map[string]func(types.NamespacedName) []string{
+			"VirtualMachine": func(_ types.NamespacedName) []string {
+				// When using vcsim additional finalizers are added.
+				if testTarget == VCSimTestTarget {
+					return []string{"virtualmachine.vmoperator.vmware.com", vcsimv1.VMFinalizer}
+				}
+				return []string{"virtualmachine.vmoperator.vmware.com"}
+			},
```

Comment:
Do we need this?

FilterObjectsWithKindAndName looks like we are filtering VIrtualMachine objects out in SupervisorTestMode

Reply:
Yes we need this, it is for vcsim vs vsphere (not supervisor vs govmomi)

Reply:
Hm. But If I understand the Filter func correctly we are never validating finalizers for VirtualMachine

* with govmomi: VirtualMachine doesn't exist
* with supervisor: we filter VirtualMachines out

I might be misinterpreting something though :)

Reply:
fixed + added some comments